src/sage/combinat/posets/bubble_and_shuffle.py

```diff
@@ -0,0 +1,203 @@
+
+
+def bubble_set(m, n) -> Iterator[tuple[int, ...]]:
+    """
+    Return the underlying set of the Bubble lattice `B_{m,n}`.
+
+    This is also the underlying set of the Shuffle lattice `S_{m,n}`.
+
+    INPUT:
+
+    - ``m`` -- integer
+    - ``n`` -- integer
+
+    EXAMPLES::
+
+        sage: from sage.combinat.posets.bubble_and_shuffle import bubble_set
+        sage: list(bubble_set(2,1))
+        [(),
+         (1,),
+         (-1,),
+         (-1, 1),
+         (1, -1),
+         (-2,),
+         (-2, 1),
+         (1, -2),
+         (-1, -2),
+         (-1, -2, 1),
+         (1, -1, -2),
+         (-1, 1, -2)]
+    """
+    X = [-i for i in range(1, m + 1)]
+    Y = list(range(1, n + 1))
+    for subX in subsets(X):
+        wX = tuple(subX)
+        for subY in subsets(Y):
+            for shu in ShuffleProduct(wX, tuple(subY)):
+                yield tuple(shu)
+
+
+def bubble_coverings(m, n, mot, transpose=True) -> Iterator[tuple[int, ...]]:
+    """
+    Return the covers of an element in the Bubble lattice `B_{m,n}`.
+
+    This can also produce covers in the Shuffle lattice `S_{m,n}`.
+
+    INPUT:
+
+    - ``m`` -- integer
+    - ``n`` -- integer
+    - ``mot`` -- element of `B_{m,n}` as a tuple
+    - ``transpose`` -- boolean (default: ``True``) whether to return covers
+      in the Bubble lattice or in the Shuffle lattice
+
+    EXAMPLES::
+
+        sage: from sage.combinat.posets.bubble_and_shuffle import bubble_coverings
+        sage: list(bubble_coverings(2, 1, (-2, 1)))
+        [(1,), (1, -2)]
+        sage: list(bubble_coverings(2, 1, (-2, 1), False))
+        [(1,)]
+    """
+    # suppression d'un x
+    for j, letter in enumerate(mot):
+        if letter < 0:
+            yield tuple(mot[:j] + mot[j + 1:])
+
+    # insertion d'un y
+    for j in range(len(mot) + 1):
+        debut, fin = tuple(mot[:j]), tuple(mot[j:])
+        yavant = max((letter for letter in debut if letter > 0), default=0)
+        yapres = min((letter for letter in fin if letter > 0), default=n + 1)
+        for y in range(yavant + 1, yapres):
+            yield debut + (y,) + fin
+
+    if not transpose:
+        return
+
+    # echange de xy vers yx
+    for j in range(len(mot) - 1):
+        if mot[j] < 0 and mot[j + 1] > 0:
+            mot2 = list(mot)
+            mot2[j] = mot[j + 1]
+            mot2[j + 1] = mot[j]
+            yield tuple(mot2)
+
+
+def BubblePoset(m, n) -> LatticePoset:
+    """
+    Return the Bubble lattice `B_{m,n}`.
+
+    Bubble lattices were introduced by McConville and Mühle in [MacCM2022]_.
+
+    EXAMPLES::
+
+        sage: P = posets.BubblePoset(2,1); P
+        Finite lattice containing 12 elements
+        sage: P.zeta_polynomial()
+        1/40*q^5 + 7/24*q^4 + 23/24*q^3 - 7/24*q^2 + 1/60*q
+    """
+    bubbles = list(bubble_set(m, n))
+
+    dg = DiGraph([(x, y) for x in bubbles for y in bubble_coverings(m, n, x)])
+    return LatticePoset(dg)
```

Comment:
```suggestion
    return LatticePoset(dg, cover_relations=True)
```
since this is only the cover relations, right? This is faster IIRC.